yolox-burn/src/model/blocks.rs

```diff
@@ -0,0 +1,327 @@
+pub fn expand(num_channels: usize, factor: f64) -> usize {
+    (num_channels as f64 * factor).floor() as usize
+}
+
+/// A base convolution block.
+/// Allows to switch between regular and depthwise separable convolution blocks based on the
+/// architecture.
+#[derive(Module, Debug)]
+pub enum Conv<B: Backend> {
+    BaseConv(BaseConv<B>),
+    DwsConv(DwsConv<B>),
+}
+
+impl<B: Backend> Conv<B> {
+    pub fn forward(&self, x: Tensor<B, 4>) -> Tensor<B, 4> {
+        match self {
+            Self::BaseConv(conv) => conv.forward(x),
+            Self::DwsConv(conv) => conv.forward(x),
+        }
+    }
+}
+
+#[derive(Config)]
+pub struct ConvConfig {
+    in_channels: usize,
+    out_channels: usize,
+    kernel_size: usize,
+    stride: usize,
+    depthwise: bool,
+}
+
+impl ConvConfig {
+    /// Initialize a new [convolution block](Conv) module.
+    pub fn init<B: Backend>(&self, device: &Device<B>) -> Conv<B> {
+        if self.depthwise {
+            Conv::DwsConv(
+                DwsConvConfig::new(
+                    self.in_channels,
+                    self.out_channels,
+                    self.kernel_size,
+                    self.stride,
+                )
+                .init(device),
+            )
+        } else {
+            Conv::BaseConv(
+                BaseConvConfig::new(
+                    self.in_channels,
+                    self.out_channels,
+                    self.kernel_size,
+                    self.stride,
+                    1,
+                )
+                .init(device),
+            )
+        }
+    }
+
+    /// Initialize a new [convolution block](Conv) module with a [record](ConvRecord).
+    pub fn init_with<B: Backend>(&self, record: ConvRecord<B>) -> Conv<B> {
+        match record {
+            ConvRecord::DwsConv(rec) if self.depthwise => Conv::DwsConv(
+                DwsConvConfig::new(
+                    self.in_channels,
+                    self.out_channels,
+                    self.kernel_size,
+                    self.stride,
+                )
+                .init_with(rec),
+            ),
+            ConvRecord::BaseConv(rec) if !self.depthwise => Conv::BaseConv(
+                BaseConvConfig::new(
+                    self.in_channels,
+                    self.out_channels,
+                    self.kernel_size,
+                    self.stride,
+                    1,
+                )
+                .init_with(rec),
+            ),
+            _ => panic!("Invalid record for depthwise={}", self.depthwise),
+        }
+    }
+}
+
+/// A Conv2d -> BatchNorm -> activation block.
+#[derive(Module, Debug)]
+pub struct BaseConv<B: Backend> {
+    conv: Conv2d<B>,
+    bn: BatchNorm<B, 2>,
+}
+
+impl<B: Backend> BaseConv<B> {
+    pub fn forward(&self, x: Tensor<B, 4>) -> Tensor<B, 4> {
+        let x = self.conv.forward(x);
+        let x = self.bn.forward(x);
+
+        // SiLU not in nn yet
```

Comment:
is this comment "// SiLU not in nn yet" still valid?

Reply:
Silu is not implemented as a module like Relu or Gelu in burn. But I can definitely remove that comment, it doesn't serve much of a purpose.